cdist/integration.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# 2017 Darko Poljak (darko.poljak at gmail.com)
+#
+# This file is part of cdist.
+#
+# cdist is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cdist is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with cdist. If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+import cdist
+import cdist.log
+import cdist.config
+import cdist.install
+import cdist.argparse
+import sys
+import os.path
+import collections
+
+
+ACTION_CONFIG = 'config'
+ACTION_INSTALL = 'install'
+
+
+def _process_hosts_simple(action, host, manifest, verbose):
+    if isinstance(host, str):
+        hosts = [host, ]
+    elif isinstance(host, collections.Iterable):
+        hosts = host
+    else:
+        raise cdist.Error('Invalid host argument: {}'.format(host))
+
+    # Setup sys.argv[0] since cdist relies on command line invocation.
+    mydir = os.path.dirname(__file__)
+    cdist_bin = os.path.abspath(os.path.join(mydir, '..', 'scripts', 'cdist'))
+    sys.argv[0] = cdist_bin
```

Comment:
Because in exec/local.py self.exec_path is set to sys.argv[0] and self.exec_path is used for emulator links:
<pre>
    def _link_types_for_emulator(self):
        """Link emulator to types"""
        src = os.path.abspath(self.exec_path)
        for cdist_type in core.CdistType.list_types(self.type_path):
            dst = os.path.join(self.bin_path, cdist_type.name)
            self.log.trace("Linking emulator: %s to %s", src, dst)

            try:
                os.symlink(src, dst)
            except OSError as e:
                raise cdist.Error(
                        "Linking emulator from %s to %s failed: %s" % (
                            src, dst, e.__str__()))
</pre>